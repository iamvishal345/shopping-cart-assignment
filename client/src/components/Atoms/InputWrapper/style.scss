@mixin label-active {
  top: -5px;
  font-size: 11px;
  left: 0px;
  color: #62c9d8;
}
.form-input {
  position: relative;
  width: 98%;
  height: 1rem;
  padding: 20px 0;
  input {
    width: 100%;
    border: none;
    border-bottom: 1px solid #989290;
    &:focus {
      outline: none;
      border-bottom: 2px solid #62c9d8;
    }
  }
  .form-input-label {
    font-size: 12px;
    position: absolute;
    pointer-events: none;
    left: 0px;
    top: 15px;
    transition: 300ms ease all;
    color: #79716f;
    &.input-value {
      @include label-active();
    }
  }
  > input:focus {
    ~ .form-input-label {
      @include label-active();
    }
  }
}
