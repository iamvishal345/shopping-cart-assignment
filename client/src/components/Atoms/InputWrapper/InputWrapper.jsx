import React from "react";
import "./style.scss";

export const InputWrapper = ({ value, setValue, name, id, label, type }) => (
  <div className="form-input">
    <input
      type={type || "text"}
      name={name}
      id={id}
      value={value}
      onChange={(e) => setValue(e.target.value.trimStart())}
    />
    <label
      className={`form-input-label ${value && "input-value"}`}
      htmlFor={id}
    >
      {label}
    </label>
  </div>
);

// InputField.propTypes = {
//   value: PropTypes.string.isRequired,
//   setValue: PropTypes.func.isRequired,
//   name: PropTypes.string.isRequired,
//   id: PropTypes.string.isRequired,
//   label: PropTypes.string.isRequired,
//   type: PropTypes.string,
//   ariaLabel: PropTypes.string,
// };
