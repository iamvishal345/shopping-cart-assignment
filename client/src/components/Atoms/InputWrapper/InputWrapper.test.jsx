import React from "react";
import { shallow } from "enzyme";
import { InputWrapper } from "./InputWrapper";

const props = {
  value: "test",
  setValue: jest.fn(),
  name: "mockName",
  id: "mockId",
  label: "mockLabel",
  type: "text",
};

let wrapped = shallow(<InputWrapper {...props} />);

describe("InputWrapper", () => {
  it("should match the snapshot", () => {
    expect(wrapped).toMatchSnapshot();
  });

  it("should call callback on  change input value", () => {
    wrapped
      .find("#mockId")
      .simulate("change", { target: { value: "modifiedValue" } });
    expect(props.setValue).toBeCalledWith("modifiedValue");
  });
});
