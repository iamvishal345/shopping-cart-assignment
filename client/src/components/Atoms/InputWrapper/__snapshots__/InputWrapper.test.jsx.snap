// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputWrapper should match the snapshot 1`] = `
<div
  className="form-input"
>
  <input
    id="mockId"
    name="mockName"
    onChange={[Function]}
    type="text"
    value="test"
  />
  <label
    className="form-input-label input-value"
    htmlFor="mockId"
  >
    mockLabel
  </label>
</div>
`;
